### PR TITLE
*: metrics by namespaces.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: server/store_statistics.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	metrics["storage_size"] = float64(s.StorageSize)
+	metrics["storage_capacity"] = float64(s.StorageCapacity)
+
+	for label, value := range metrics {
+		clusterStatusGauge.WithLabelValues(label, s.namespace).Set(value)
+	}
+}
+
+// It is used for counting up stats of all namespaces.
+const allNamespace = "_all"
```
Comment (siddontang) [inline server/store_statistics.go]: seem we can aggregate all metrics in Prometheus directly.

File: server/store_statistics.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	if !ok {
+		stat = newStoreStatistics(m.opt, namespace)
+		m.stats[namespace] = stat
+	}
+	stat.Observe(store)
+	m.all.Observe(store)
+}
+
+func (m *storeStatisticsMap) Collect() {
+	m.all.Collect()
```
Comment (siddontang) [inline server/store_statistics.go]: need Lock here?

File: server/store_statistics.go (hunk 1 above)
Comment (disksing) [reply server/store_statistics.go]: I want to use templating to select namespace, but the templating query supports very limited queries. I found no approach to join a constant with a query result vector. /cc @huachaohuang

File: server/store_statistics.go (hunk 2 above)
Comment (disksing) [reply server/store_statistics.go]: No.

File: server/store_statistics.go (hunk 1 above)
Comment (siddontang) [reply server/store_statistics.go]: If we can't do it in Grafana, maybe we can create a new board for all.

**Comment by disksing** [comment]
PTAL @nolouch @siddontang

**Comment by siddontang** [comment]
LGTM

**Pull request merged**